packages/manager/apps/pci-gateway/src/api/hooks/operation.ts

```diff
@@ -0,0 +1,54 @@
+import { v6 } from '@ovh-ux/manager-core-api';
+import { useEffect, useState } from 'react';
+import { TOperation } from '@/api/data/operation';
+
+type TState = {
+  opId: string;
+  operation: TOperation;
+  isPending: boolean;
+};
+
+const getOperation = async (projectId: string, operationId: string) => {
+  const { data } = await v6.get<TOperation>(
+    `/cloud/project/${projectId}/operation/${operationId}`,
+  );
+  return data;
+};
+
+export const useOperation = ({
+  projectId,
+  operationId,
+  interval = 2000,
+  onSuccess = () => {},
+}: {
+  projectId: string;
+  operationId: string;
+  interval?: number;
+  onSuccess: () => void;
+}) => {
+  const [state, setState] = useState<TState>({
+    opId: undefined,
+    operation: undefined,
+    isPending: false,
+  });
+
+  let intervalId: NodeJS.Timer;
+
+  useEffect(() => {
+    if (projectId && operationId) {
+      setState((prev) => ({ ...prev, isPending: true }));
+      if (intervalId) clearInterval(intervalId);
+      intervalId = setInterval(async () => {
+        const operation = await getOperation(projectId, operationId);
+        setState((prev) => ({ ...prev, operation }));
+        if (operation.status === 'completed') {
```

Comment:
i think operation could fails so you need to handle the case too otherwise it'll never end polling

i would recommend using setTimeout instead of setInterval because if a network request takes time to resolve you can flood the network calls stack with a setInterval, with setTimeout you can call the function again when the network calls resolves